code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java

```diff
@@ -200,6 +203,39 @@ public static void log(@NonNull final LoggingMode mode, final String tag, final
         }
     }
 
+    public static void registerExtensions(@NonNull final List<Class<? extends Extension>> extensions, final AdobeCallback<?> completionCallback) {
```

Comment:
What happens if:

a. this function is called more than once?  (at face value the registeredExtensions atomic int is going to cause some problems because it's initialized to zero)

b. if this function is called more than once across threads? (less important but still potentially a problem)

Reply:
@pfransenadb This function behaves similarly to the one we have in Swift SDK. It registers the extensions passed to it with the event hub, starts event processing and then notifies the caller by calling `completionCallback`. 

We expect customers to simplify SDK initialization by calling new registerExtensions() API once although calling it multiple times will work properly. 

a. Each call to `registerExtensions` will call the `completionCallback` after registering the extensions passed to it with EventHub. `registeredExtensions` atomic int is a local variable and only tracks extensions passed to a particular call. 

b. EventHub is thread safe and processes everything in its own queue. Calling from different threads should work as expected.